pkg/virtual/apiexport/authorizer/maximal_permission_policy.go

```diff
@@ -0,0 +1,176 @@
+func NewMaximalPermissionAuthorizer(deepSARClient kubernetes.ClusterInterface, apiExportInformer apisinformers.APIExportInformer, apiBindingInformer apisinformers.APIBindingInformer) authorizer.Authorizer {
+	apiExportLister := apiExportInformer.Lister()
+	apiExportIndexer := apiExportInformer.Informer().GetIndexer()
+
+	return &maximalPermissionAuthorizer{
+		getAPIExport: func(clusterName, apiExportName string) (*apisv1alpha1.APIExport, error) {
+			return apiExportLister.Get(clusters.ToClusterAwareKey(logicalcluster.New(clusterName), apiExportName))
+		},
+		getAPIExportsByIdentity: func(identityHash string) ([]*apisv1alpha1.APIExport, error) {
+			return indexers.ByIndex[*apisv1alpha1.APIExport](apiExportIndexer, indexers.APIExportByIdentity, identityHash)
+		},
+		newDeepSARAuthorizer: func(clusterName logicalcluster.Name) (authorizer.Authorizer, error) {
+			return delegated.NewDelegatedAuthorizer(clusterName, deepSARClient)
+		},
+	}
+}
+
+func (a *maximalPermissionAuthorizer) Authorize(ctx context.Context, attr authorizer.Attributes) (authorizer.Decision, string, error) {
+	apiDomainKey := dynamiccontext.APIDomainKeyFrom(ctx)
+	parts := strings.Split(string(apiDomainKey), "/")
+	if len(parts) < 2 {
+		return authorizer.DecisionNoOpinion, VirtualAPIExportNotPermittedReason, fmt.Errorf("invalid API domain key")
+	}
+
+	claimingAPIExportCluster := parts[0]
+	claimingAPIExportName := parts[1]
+
+	claimingAPIExport, err := a.getAPIExport(claimingAPIExportCluster, claimingAPIExportName)
+	if kerrors.IsNotFound(err) {
+		return authorizer.DecisionNoOpinion, VirtualAPIExportNotPermittedReason, fmt.Errorf("API export not found: %w", err)
+	}
+
+	claimedIdentityHash, found := getClaimedIdentity(claimingAPIExport, attr)
+	if !found {
+		// it's a resource in the claiming API export, hence unclaimed
+		return authorizer.DecisionAllow, "unclaimed resource", nil
+	}
+	if claimedIdentityHash == "" {
+		// it's a native k8s resource (secret, configmap, ...), or a system kcp CRD resource (apis.kcp.dev)
+		// For neither case a maximum permission policy can exist.
+		return authorizer.DecisionAllow, "unclaimable resource", nil
+	}
+
+	claimedAPIExports, err := a.getAPIExportsByIdentity(claimedIdentityHash)
+	if err != nil {
+		return authorizer.DecisionNoOpinion, VirtualAPIExportNotPermittedReason, fmt.Errorf("error getting API export identity: %q: %w", claimedIdentityHash, err)
+	}
+
+	if len(claimedAPIExports) == 0 {
+		// a claimed identity hash exists but not API export can be found referring to it (potentially eventually consistent).
+		// In this case be safe and deny the request, forcing the caller to retry.
+		return authorizer.DecisionDeny, VirtualAPIExportNotPermittedReason, nil
+	}
+
+	// multiple claimed API exports can share the same identity hash (even in different workspaces).
+	// All maximum permission policies must grant access because at this point no deterministic API export can be picked
+	// (especially in the wildcard case).
```

Comment:
I thought the GoDoc above said we don't even bother to check in wildcard?

Reply:
I think the brackets can be indeed left out as the code doesn't make a difference any more between wildcard and non-wildcard requests.

Reply:
removed the brackets and the wildcard case sentence